### PR TITLE
sales round-robin with location

### PR DESCRIPTION
**Pull request by PrenSJ2**
in `choose-sales-person`﻿ we now get the location from the header

- also added sells gb, eu, us, au, row to Admin model

### to test
- run make restore-from-live
- run uvicorn
- run aerich upgrade
- navigate to localhost:8000/login (after the restore you should be authorized)
- then navigate to /admin and edit an admin to have sells GB
- then navigate to http://localhost:8000/choose-roundrobin/sales/?plan=payg and get that admin

### CODE REVIEW

**Comment by PrenSJ2** [comment]
after this is merged then this should be aswell https://github.com/tutorcruncher/tutorcruncher.com/pull/2160

**Review by DanJezeph15 — COMMENTED** [review]
Ignore some of these comments if its fine, I'm not really sure the intended way for each thing to work.

* [ ] Error from earlier when no one set to sell anywhere.

```
File "/home/dan/repos/hermes/app/hermes/views.py", line 112, in choose_sales_person
    next_sales_person = admin_ids[0]
```

* [ ] You mentioned what happens if you go to choose-roundrobin with no plan and you get the following:
image

* [ ] Does `cf_ipcountry` definitely work? If I print out what the value for it is, I always seem to get None. If it does work, any chance you can explain it to me.

* [ ] We should have a fallback for the following scenario:
Fionn set up to sell enterprise and nothing else, with sell_us, sell_gb, sell_row, etc all true
Sam setup to sell payg and startup but only in the uk
What happens if we have a payg coming in from the US. We'd have no one to take the call as Fionn doesn't sell that level and Sam who does is only setup for the UK

**Comment by PrenSJ2** [comment]
CF-IPCountry - is a request header appended by cloudflare, so we cant test it locally

to test live use a vpn

File: app/hermes/views.py [hunk 1]
```diff
@@ -21,17 +85,47 @@ async def choose_sales_person(plan: str) -> Admin.pydantic_schema():
...
     else:
-        next_sales_person = admin_ids[0]
+        if latest_company:
+            latest_sales_person = latest_company.sales_person_id
+            try:
+                next_sales_person = admins_ids[admins_ids.index(latest_sales_person) + 1]
+            except (IndexError, ValueError):
+                next_sales_person = admins_ids[0]
+        else:
+            next_sales_person = admins_ids[0]
```
Comment (DanJezeph15) [inline app/hermes/views.py:127]: This function is doing too much now imo and both parts use similar logic, I think for readability you should have a function called like `get_next_sales_person` and then it should take the admins and the latest_company if it exists. So it looks something like 

```python
if regional_admins:
    next_sales_person = _get_next_sales_person(regional_admins)
else:
    next_sales_person = _get_next_sales_person(admins)
```

and then the func could do something like:
```python
def _get_next_sales_person(admins, latest_company):
    if latest_company:
        <your try except code using the admins list passed through>
    else:
        <return like admins[0] or something>
```

Play around with the func tho and make sure it works as thats just pseudo code, then you can add a docstring and typehint so its easier to understand whats going on too.

**Review by DanJezeph15 — CHANGES_REQUESTED** [review]

**Pull request merged**